src/shared/tests/pipes.test.tsx

```diff
@@ -0,0 +1,247 @@
+  });
+  test('with octal', () => {
+    expect(weeksPipe(0o377)).toBe('255 weeks');
+  });
+  test('with binary', () => {
+    expect(weeksPipe(0b111001)).toBe('57 weeks');
+  });
+});
+
+describe('Formatting Dollars Tests', () => {
+  test('with 1', () => {
+    expect(dollarsPipe(1)).toBe('$1');
+  });
+  test('with int > 1', () => {
+    expect(dollarsPipe(2)).toBe('$2');
+  });
+  test('with 0', () => {
+    expect(dollarsPipe(0)).toBe('$0');
+  });
+  test('with decimal', () => {
+    expect(dollarsPipe(6.9)).toBe('$6.9');
+    expect(dollarsPipe(4.2)).toBe('$4.2');
+  });
+  test('with hexadecimal', () => {
+    expect(dollarsPipe(0x37cf)).toBe('$14287');
+  });
+  test('with octal', () => {
+    expect(dollarsPipe(0o377)).toBe('$255');
+  });
+  test('with binary', () => {
+    expect(dollarsPipe(0b111001)).toBe('$57');
+  });
+});
+
+describe('Formatting Booleans', () => {
+  test('Formatting booleans', () => {
+    expect(booleanPipe(true)).toBe('YES');
+    expect(booleanPipe(false)).toBe('NO');
+  });
+});
+
+describe('Formatting Empty Strings Without Dash', () => {
+  test('empty string', () => {
+    expect(emptyStringPipe('')).toBe('');
+  });
+  test('empty string with space', () => {
+    expect(emptyStringPipe(' ')).toBe(' ');
+  });
+  test('non-empty string', () => {
+    expect(emptyStringPipe('abcd')).toBe('abcd');
+  });
+  test('non-empty string with numbers', () => {
+    expect(emptyStringPipe('123')).toBe('123');
+  });
+  test('non-empty string with special characters', () => {
+    expect(emptyStringPipe('!@#$%^&*()}{:">?')).toBe('!@#$%^&*()}{:">?');
+  });
+});
+
+describe('Formatting Empty Strings With Dash', () => {
+  test('empty string', () => {
+    expect(emDashPipe('')).toBe('');
+  });
+  test('empty string with space', () => {
+    expect(emDashPipe(' ')).toBe(' ');
+  });
+  test('non-empty string', () => {
+    expect(emDashPipe('abcd')).toBe('abcd');
+  });
+  test('non-empty string with numbers', () => {
+    expect(emDashPipe('123')).toBe('123');
+  });
+  test('non-empty string with special characters', () => {
+    expect(emDashPipe('!@#$%^&*()}{:">?')).toBe('!@#$%^&*()}{:">?');
+  });
+});
+
+describe('Formatting Dates', () => {
+  test('invalid test', () => {
+    expect(datePipe(new Date(99999999999999999999))).toBe('Invalid Date');
+  });
+  test('empty constructor/current time', () => {
+    let current_time = new Date();
+    let answer = current_time.toLocaleString('en-US');
+    expect(datePipe(new Date())).toBe(answer);
+  });
+  // This test fails on github but passes locally.
+  test.skip('milliseconds', () => {
+    expect(datePipe(new Date(1000000000))).toBe('1/12/1970, 8:46:40 AM');
+  });
+  test('datestring', () => {
+    expect(datePipe(new Date('2021-10-5'))).toBe('10/5/2021, 12:00:00 AM');
+  });
+  test('custom detailed', () => {
+    expect(datePipe(new Date(2021, 0o11, 0o5, 14, 3, 6, 9))).toBe('10/5/2021, 2:03:06 PM');
+  });
+});
+
+describe('Formatting Wbs Numbers', () => {
```

Comment:
Let's add a test case that doesn't use the dummy data so we can have 2 and 3 digit numbers in the WBS#